closed/custom/Docs.gmk

```diff
@@ -103,18 +100,214 @@ OPENJ9_JAVADOC_SCRIPT_BODY = \
+    for file in `$(FIND) $(OPENJ9_ONLY_DOCDIR) -type f` ; do \
+      $(CP) -f "$$file" "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+      $(call sed_replace_all,class="external-link",class="external-link" target="_blank") <"$(OPENJ9_ONLY_JAVADOC_TEMP)" >"$$file" ; \
+    done ; \
+    for file in `cd $(OPENJ9_ONLY_DOCDIR) ; $(FIND) . -type f -name module-summary.html | $(SED) -e 's|^\./||'` ; do \
+      fullpath=$(OPENJ9_ONLY_DOCDIR)/$$file ; \
+      dir="$${file%/*}" ; \
+      topdir="$${file%%/*}" ; \
+      excepttopdir="$${file\#*/}" ; \
+      excepttopdir=`$(SED) 's|-summary.html$$|-info.java|' <<<$$excepttopdir` ; \
+      regex="$$topdir/.*$$excepttopdir" ; \
+      if ! $(GREP) -qe "$$regex" $(OPENJ9_ONLY_JAVADOC_FILES) ; then \
+        $(CP) -f "$$fullpath" "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+        $(call sed_replace_block,$(MODULE_DESCRIPTION_FIRST_LINE),$(MODULE_DESCRIPTION_END_LINE),$(MODULE_DESCRIPTION_REPLACE_TEXT)) <"$(OPENJ9_ONLY_JAVADOC_TEMP)" >"$$fullpath"  ; \
+        $(CP) -f "$$fullpath" "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+        $(call sed_replace_block,$(MODULE_SERVICES_FIRST_LINE),$(MODULE_SERVICES_END_LINE),$(MODULE_SERVICES_REPLACE_TEXT)) <"$(OPENJ9_ONLY_JAVADOC_TEMP)" >"$$fullpath"  ; \
+        $(RM) "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+      fi ; \
+    done ; \
+    if ! $(GREP) -q '$(OPENJDK_JDK_MANAGEMENT_SUMMARY)' $(OPENJ9_ONLY_INDEX_HTML) ; then \
+      $(ECHO) "$$0: $(OPENJDK_JDK_MANAGEMENT_SUMMARY_ERROR_TEXT)" ; \
+      exit 1 ; \
+    else \
+      $(CP) -f "$(OPENJ9_ONLY_INDEX_HTML)" "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+      $(call sed_replace_first,$(OPENJDK_JDK_MANAGEMENT_SUMMARY),$(OPENJDK_MODULE_REFERENCE_TEXT)) <"$(OPENJ9_ONLY_JAVADOC_TEMP)" >"$(OPENJ9_ONLY_INDEX_HTML)" ; \
+      $(RM) "$(OPENJ9_ONLY_JAVADOC_TEMP)" ; \
+    fi ; \
+    #
+
+OPENJ9_ONLY_JAVADOC_SCRIPT_BODY_ESC = $(subst ','\'',$(OPENJ9_ONLY_JAVADOC_SCRIPT_BODY))
+
+# Create a stub package-info.java file for each package with source files in
+# j9jcl_sources but no package-info.java file. This is then used by javadoc
+# instead of the openjdk version to direct the user to the Oracle online
+# documentation for that package.
+
+# The dependency on $(OPENJDK_JAVADOC_MARKER) ensures that the docs/api/element-list
+# file which is created during the full jdk javadoc generation exists before running
+# javadoc for just the openj9 files.
+
+# If running on Windows, the source file names in the response file must be converted to
+# Windows format before passing them to javadoc.
+OPENJ9_ONLY_JAVADOC_FIXPATH_FILELIST := $(foreach file,$(OPENJ9_ONLY_JAVADOC_FILELIST),$(call FixPath,$(file)))
+
+$(OPENJ9_ONLY_JAVADOC_MARKER) : $(OPENJDK_JAVADOC_MARKER) $(OPENJ9_ONLY_FILELIST)
+	$(call LogWarn, Generating javadoc for openj9 source files)
+	$(eval $(call ListPathsSafely,OPENJ9_ONLY_JAVADOC_FIXPATH_FILELIST,$(OPENJ9_ONLY_JAVADOC_FILES)))
+	for dir in $(OPENJDK_ONLY_PKGLIST) ; do \
+ 		$(MKDIR) -p $(OPENJ9_STUBS_DIR)/$${dir} ; \
```

Comment:
Please remove the leading space (before the first tab) to match jdk16.
Then squash.

Reply:
Whitespace removed and squashed.